actors/migration/nv16/top.go

```diff
@@ -66,38 +66,62 @@ func ActorHeadKey(addr address.Address, head cid.Cid) string {
 
 // Migrates the filecoin state tree starting from the global state tree and upgrading all actor state.
 // The store must support concurrent writes (even if the configured worker count is 1).
-func MigrateStateTree(ctx context.Context, store cbor.IpldStore, actorsRootIn cid.Cid, priorEpoch abi.ChainEpoch, cfg Config, log Logger, cache MigrationCache) (cid.Cid, error) {
+func MigrateStateTree(ctx context.Context, store cbor.IpldStore, actorsManifest cid.Cid, actorsRootIn cid.Cid, priorEpoch abi.ChainEpoch, cfg Config, log Logger, cache MigrationCache) (cid.Cid, error) {
 	if cfg.MaxWorkers <= 0 {
 		return cid.Undef, xerrors.Errorf("invalid migration config with %d workers", cfg.MaxWorkers)
 	}
 
+	adtStore := adt8.WrapStore(ctx, store)
+
+	// load manifest
+	var manifest manifest8.Manifest
+	if err := adtStore.Get(ctx, actorsManifest, &manifest); err != nil {
```

Comment:
bike shed: I'm not convinced of the value of the separate versioned manifest object that only comes in during the migration and doesn't make it on chain.  If we want the version for the migration why not on chain?  Even better, if we don't care about the version on chain why include it in the migration at all? I don't see the point of the version and outer wrapper why not just pass the manifest data cid.

Reply:
At the end of the day its a few bytes so nbd

Reply:
two things here:
- caller doesnt have to parse
- we need to know the version to parse

So i think its much cleaner to pass the cid to the whole manifest to us  as opposed to complicating callers and having to pass the version as well.
And if we ever change the manifest format, user code wont have to change.

Reply:
Why we aren't keeping the version on chain too?

Reply:
because we dont have to?
The data format, and hence the state of the system actor, is imploed by the network version.

Reply:
In theory, the version on the manifest is the version of the **exchange, portable format** (i.e. how the manifest is encoded in the bundle). It is entirely possible for the structure on chain to diverge from the bundle structure. The structure on chain would evolve through network versions; the structure in the bundle would evolve through the manifest version.

The code in this PR has some implicit coupling between the two, such as the system actor state constructor forming an empty `ManifestData`, or the getter loading a `ManifestData`, but it's not too harmful given that specs-actors only exists from now onwards to support migrations and provide Go types.